### PR TITLE
Issue #14937: Migrated section 4.8.3.2 No C Style Array Declarations to chapter wise testing

### PR DESCRIPTION
**Pull request by Zopsss**
#14937

### CODE REVIEW

File: src/it/java/com/google/checkstyle/test/chapter4formatting/rule4832nocstylearray/NoCStyleArrayDeclarationsTest.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.google.checkstyle.test.chapter4formatting.rule4832nocstylearray;
+
+import org.junit.jupiter.api.Test;
+
+import com.google.checkstyle.test.base.AbstractGoogleModuleTestSupport;
+
+// -@cs[AbbreviationAsWordInName] Test should be named as its chapter name.
+public class NoCStyleArrayDeclarationsTest extends AbstractGoogleModuleTestSupport {
```
Comment (Zopsss) [inline src/it/java/com/google/checkstyle/test/chapter4formatting/rule4832nocstylearray/NoCStyleArrayDeclarationsTest.java:27]: will removed the suppression once https://github.com/checkstyle/checkstyle/pull/15116#discussion_r1652192468 is merged

Comment (romani) [reply src/it/java/com/google/checkstyle/test/chapter4formatting/rule4832nocstylearray/NoCStyleArrayDeclarationsTest.java:27]: I thinks this `CStyle` should be ok to hardcode at `getExtractedRuleName(String ruleName)` and known edge case that we simply do search/replace to `Cstyle` before ` return extractedRuleName.toString();`

**Review by romani — CHANGES_REQUESTED** [review]
see above

File: src/it/java/com/google/checkstyle/test/chapter4formatting/rule4832nocstylearray/NoCStyleArrayDeclarationsTest.java (hunk 1 above)
Comment (romani) [reply src/it/java/com/google/checkstyle/test/chapter4formatting/rule4832nocstylearray/NoCStyleArrayDeclarationsTest.java]: I pushed an update

**Review by romani — APPROVED** [review]
ok to merge

File: src/it/java/com/google/checkstyle/test/chapter4formatting/rule4832nocstylearray/NoCStyleArrayDeclarationsTest.java (hunk 1 above)
Comment (Zopsss) [reply src/it/java/com/google/checkstyle/test/chapter4formatting/rule4832nocstylearray/NoCStyleArrayDeclarationsTest.java]: I'll remove the suppression once the update has been merged

Comment (Zopsss) [reply src/it/java/com/google/checkstyle/test/chapter4formatting/rule4832nocstylearray/NoCStyleArrayDeclarationsTest.java]: removed the suppression:

**Review by rdiachenko — APPROVED** [review]
lgtm

**Comment by rdiachenko** [comment]
Rebased

**Pull request merged**